sklearn/inspection/_plot/decision_boundary.py

```diff
@@ -0,0 +1,193 @@
+        return getattr(estimator, 'predict_proba')
+    elif hasattr(estimator, 'decision_function'):
+        return getattr(estimator, 'decision_function')
+
+    raise ValueError(error_msg.format(
+        "decision_function or predict_proba",
+        estimator.__class__.__name__))
+
+
+class DecisionBoundaryDisplay:
+    """Decisions Boundary visualization.
+
+    It is recommend to use :func:`~sklearn.inspection.plot_decision_boundary`
+    to create a :class:`DecisionBoundaryDisplay`. All parameters are stored as
+    attributes.
+
+    Read more in the :ref:`User Guide <visualizations>`.
+
+    Parameters
+    ----------
+    xx0 : ndarray of shape (grid_resolution, grid_resolution)
+        First output of meshgrid.
+
+    xx1 : ndarray of shape (grid_resolution, grid_resolution)
+        Second output of meshgrid.
+
+    response : ndarray of shape (grid_resolution, grid_resolution)
+        Values of the response function.
+
+    Attributes
+    ----------
+    surface_ : matplotlib `QuadContourSet` or `QuadMesh`
+        If `plot_method` is 'contour' or 'contourf', `surface_` is a
+        `QuadContourSet`. If `plot_method is `pcolormesh`, `surface_` is a
+        `QuadMesh`.
+
+    ax_ : matplotlib Axes
+        Axes with confusion matrix.
+
+    figure_ : matplotlib Figure
+        Figure containing the confusion matrix.
+    """
+    def __init__(self, xx0, xx1, response):
+        self.xx0 = xx0
+        self.xx1 = xx1
+        self.response = response
+
+    def plot(self, plot_method='contourf', ax=None, **kwargs):
+        """Plot visualization.
+
+        Parameters
+        ----------
+        plot_method : {'contourf', 'contour', 'pcolormesh'}, default='contourf'
+            Plotting method to call when plotting the response.
+
+        ax : Matplotlib axes, default=None
+            Axes object to plot on. If `None`, a new figure and axes is
+            created.
+
+        **kwargs : dict
+            Additional keyword arguments to be pased to the `plot_method`.
+
+        Returns
+        -------
+        display: :class:`~sklearn.inspection.DecisionBoundaryDisplay`
+        """
+        check_matplotlib_support('DecisionBoundaryDisplay.plot')
+        import matplotlib.pyplot as plt  # noqa
+
+        if plot_method not in ('contourf', 'contour', 'pcolormesh'):
+            raise ValueError("plot_method must be 'contourf', 'contour', or "
+                             "'pcolormesh'")
+
+        if ax is None:
+            _, ax = plt.subplots()
+
+        plot_func = getattr(ax, plot_method)
+        self.surface_ = plot_func(self.xx0, self.xx1, self.response, **kwargs)
+        self.ax_ = ax
+        self.figure_ = ax.figure
+        return self
+
+
+def plot_decision_boundary(estimator, X, grid_resolution=100, eps=1.0,
+                           plot_method='contourf',
+                           response_method='auto',
+                           ax=None, **kwargs):
+    """Plot Decision Boundary.
```

Comment:
This would need a UG link or at least something like "please see examples below for usage"